### PR TITLE
acceptanceccl: fix restore tests

### PR DESCRIPTION
**Pull request by benesch**
See individual commit messages for details.

### CODE REVIEW

**Comment by dt** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r2, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-:-KrONTHx-0MaP1rn0bx5:bnfp4nl)*

**Review by madelynnblue — APPROVED** [review]

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r1, 3 of 3 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/acceptance/terrafarm/run.go, line 62 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-KrX6CAHVTVoERERZwlM:-KrX6CAHVTVoERERZwlN:bzb72pe) ([raw file](https://github.com/cockroachdb/cockroach/blob/e057cd06b6a0ac45a5f2148ab9af85e312cbb5e3/pkg/acceptance/terrafarm/run.go#L62)):*
> ```Go
> 		"-state="+f.StateFile,
> 		`-var=prefix="`+f.Prefix+`"`)
> 	return append(args, f.TerraformArgs...)
> ```

note that this will have a merge conflict with #17457.

---

*[pkg/ccl/acceptanceccl/backup_test.go, line 281 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-KrX6S7lYuoE5HSvcy7_:-KrX6S7lYuoE5HSvcy7a:b-66u57o) ([raw file](https://github.com/cockroachdb/cockroach/blob/ef56c3961112560eca466219b9a50de0bcff9600/pkg/ccl/acceptanceccl/backup_test.go#L281)):*
> ```Go
> 	bt := benchmarkTest{
> 		b:                   b,
> 		nodes:               10,
> ```

consider adding a comment explaining this value, particularly if you predict that a different VM class may be used in the future.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624)*

**Comment by tamird** [comment]
Reviewed 3 of 3 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, 1 of 1 files at r11.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624)*

**Comment by benesch** [comment]
Ok, PTAL! Had to add a few more assorted fixes.

Here's manually-triggered runs of RestoreTPCH10 and Restore2TB:

  * https://teamcity.cockroachdb.com/viewLog.html?buildId=330550&buildTypeId=Cockroach_Nightlies_RestoreTpch10&tab=buildLog
  * https://teamcity.cockroachdb.com/viewLog.html?buildId=330552&buildTypeId=Cockroach_Nightlies_Restore2TB&tab=buildLog

/cc @danhhz)

**Comment by benesch** [comment]
Hmm, looks like 2TB causes a liveness failure after ~5% of progress, but the test seems correct, at least.

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r12, 5 of 5 files at r18, 2 of 2 files at r19, 4 of 4 files at r20, 1 of 1 files at r21.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-:-Ks9ghXGYEaBcQCRjqqK:bnfp4nl)*

**Comment by danhhz** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-:-Ks9qnpB6QlqImEo0kTh:bnfp4nl)*

**Comment by benesch** [comment]
TFTRs!

File: pkg/ccl/acceptanceccl/backup_test.go [hunk 1]
```diff
@@ -127,6 +143,13 @@ func (bt *benchmarkTest) Start(ctx context.Context) {
 	// liveness failures. This limit was determined experimentally on
 	// Standard_D3_v2 instances.
 	sqlDB.Exec(`SET CLUSTER SETTING kv.bulk_io_write.max_rate = '30MB'`)
+	// Stats-based replica and lease rebalancing interacts badly with restore's
+	// splits and scatters.
+	//
+	// TODO(benesch): Remove these settings when #17671 is fixed, or document the
+	// necessity of these settings as 1.1 known limitation.
+	sqlDB.Exec(`SET CLUSTER SETTING kv.allocator.stat_based_rebalancing.enabled = false`)
+	sqlDB.Exec(`SET CLUSTER SETTING kv.allocator.load_based_lease_rebalancing.enabled = false`)
```
Comment (a-robinson) [inline pkg/ccl/acceptanceccl/backup_test.go]: @benesch I understand stat-based rebalancing caused problems, but was this setting independently shown to be an issue?

**Pull request merged**